GeneratorInterface/Core/src/BaseHadronizer.cc

```diff
@@ -1,20 +1,154 @@
+
+#include <random>
+#include <sys/wait.h>
 
 namespace gen {
 
 const std::vector<std::string> BaseHadronizer::theSharedResources;
 
-BaseHadronizer::BaseHadronizer( edm::ParameterSet const& ps )
-{
+  BaseHadronizer::BaseHadronizer( edm::ParameterSet const& ps ) :
+  randomIndex_(-1),
+  gridpackPaths_(1)
+  {
+    
+    if (ps.exists("RandomizedParameters")) {
+      std::vector<edm::ParameterSet> randomizedParameters = ps.getParameter<std::vector<edm::ParameterSet> >("RandomizedParameters");
+      randomInitWeights_.resize(randomizedParameters.size());
+      randomInitConfigDescriptions_.resize(randomizedParameters.size());
+      gridpackPaths_.resize(randomizedParameters.size());
+      for (unsigned int irand = 0; irand<randomizedParameters.size(); ++irand) {
+        randomInitWeights_[irand] = randomizedParameters[irand].getParameter<double>("ConfigWeight");
+        if (randomizedParameters[irand].exists("ConfigDescription")) {
+          randomInitConfigDescriptions_[irand] = randomizedParameters[irand].getParameter<std::string>("ConfigDescription");
+        }
+        if (randomizedParameters[irand].exists("GridpackPath")) {
+          gridpackPaths_[irand] = randomizedParameters[irand].getParameter<std::string>("GridpackPath");
+        }
+      }
+    }
+    else {
+      if (ps.exists("GridpackPath")) {
+        gridpackPaths_[0] = ps.getParameter<std::string>("GridpackPath");
+      }
+    }
+    
+    runInfo().setFilterEfficiency(
+        ps.getUntrackedParameter<double>("filterEfficiency", -1.) );
+    runInfo().setExternalXSecLO(
+        GenRunInfoProduct::XSec(ps.getUntrackedParameter<double>("crossSection", -1.)) );
+    runInfo().setExternalXSecNLO(
+        GenRunInfoProduct::XSec(ps.getUntrackedParameter<double>("crossSectionNLO", -1.)) );
 
-   runInfo().setFilterEfficiency(
-      ps.getUntrackedParameter<double>("filterEfficiency", -1.) );
-   runInfo().setExternalXSecLO(
-      GenRunInfoProduct::XSec(ps.getUntrackedParameter<double>("crossSection", -1.)) );
-   runInfo().setExternalXSecNLO(
-       GenRunInfoProduct::XSec(ps.getUntrackedParameter<double>("crossSectionNLO", -1.)) );
+  }
+  
+  GenLumiInfoHeader *BaseHadronizer::getGenLumiInfoHeader() const {
+    
+    GenLumiInfoHeader *genLumiInfoHeader = new GenLumiInfoHeader();
+    
+    //fill information on randomized configs for parameter scans
+    genLumiInfoHeader->setRandomConfigIndex(randomIndex_);
+    if (randomIndex_>=0) {
+      genLumiInfoHeader->setConfigDescription(randomInitConfigDescription());      
+    }
+    
+    return genLumiInfoHeader;
+    
+  }
 
-}
+  void BaseHadronizer::randomizeIndex(edm::LuminosityBlock const& lumi, CLHEP::HepRandomEngine* rengine) {
+    if (randomInitWeights_.size()>0) {
+      //randomly select from a list of provided configuration sets (for parameter scans)
+      
+      //seeds std 32-bit mersene twister with HepRandomEngine state plus run and lumi section numbers
```

Comment:
@Dr15Jones this is the really final iteration of the random seed initialization where now literally the full internal state of the HepJamesRandom is used together with the run and lumi numbers to initialize the std random number generator.